File: mipush_hook/src/main/java/com/nihility/EmptyOuterDependencies.java
package com.nihility;

import com.nihility.service.XMPushServiceListener;
import com.xiaomi.push.service.XMPushService;

public class EmptyOuterDependencies implements OuterDependencies {

    @Override
    public Configurations configuration() {
        return null;
    }

    @Override
    public XMPushServiceListener serviceListener(XMPushService pushService) {
        return null;
    }

    @Override
    public HookedMethodHandler hookedMethodHandler() {
        return null;
    }
}

File: mipush_hook/src/main/java/com/nihility/EmptyOuterDependencies.kt
package com.nihility

import com.nihility.service.XMPushServiceListener
import com.xiaomi.push.service.XMPushService

class EmptyOuterDependencies : OuterDependencies {
    override fun configuration(): Configurations? = null
    override fun serviceListener(pushService: XMPushService): XMPushServiceListener? = null
    override fun hookedMethodHandler(): HookedMethodHandler? = null
}
